src/archemist/core/optimisation/optimisation_handler.py

```diff
@@ -0,0 +1,59 @@
+import time
+from archemist.core.persistence.recipe_files_watchdog import RecipeFilesWatchdog
+from archemist.core.persistence.persistence_manager import PersistenceManager
+from pathlib import Path
+from threading import Thread
+import pandas as pd
+
+
+class OptimizationHandler:
+    def __init__(self, recipe_dir, max_recipe_count, optimizer, optimization_state,recipe_name) -> None:
+        self._recipe_path = recipe_dir
+        self._recipe_name = recipe_name
+        self._result_path = Path.joinpath(self._recipe_path, "result")
+        self._max_number_of_recipes = max_recipe_count
+        self._optimizer = optimizer
+        self._optimization_state = optimization_state
+        self._opt_update_dict = {'CSCSVJobOpDescriptor':{'dispense_info':{}},
+            'SampleColorOpDescriptor': {'red_intensity': ''}# ,
+                                                            #  'green_intensity': '',
+                                                            #  'blue_intensity': ''}
+                                 } # to get a formula that converts 3 values into one
+
+        host = 'mongodb://localhost:27017'
+        pm = PersistenceManager(host, 'algae_bot_workflow')
+        self._state = pm.construct_state_from_db()
```

Comment:
you can just pass the state in the constructor instead of doing this here. Please remove this and pass State as one of the parameters for this class constructor. You will also want to pass the State to the optimisation manager so it can pass it along